src/engine/Server.cpp

```diff
@@ -977,6 +977,9 @@ Awaitable<void> Server::processQueryOrUpdate(
     TimeLimit timeLimit) {
   using namespace ad_utility::httpUtils;
 
+  static_assert(std::is_same_v<Operation, Query> ||
+                std::is_same_v<Operation, Update>);
```

Comment:
I think we have something like `ad_utility::SameAsAny<Operation, Query, Update>` in the `util/TypeTraits.h` header.